src/smart-components/OpenchannelConversation/hooks/useTrimMessageList.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState } from 'react';
+import * as messageActionTypes from '../dux/actionTypes';
+
+interface DynamicParams {
+  messagesLength: number;
+  experimentalMessageLimit: number;
+}
+
+interface StaticParams {
+  messagesDispatcher: ({
+    type: string,
+    payload: { experimentalMessageLimit: number }
+  }) => void;
+  logger: SendbirdUIKit.Logger;
+}
+
+const THROTTLE_TIMER = 3000;
+
+// to trim message list so that we wont keep thousands of messages in memory
+// We are throttling here; not debouncing
+// it will be called once very 5 sec if messagesLength, experimentalMessageLimit changes
+// we check if messagesLength > experimentalMessageLimit before dispatching action
+function useTrimMessageList(
+  { messagesLength, experimentalMessageLimit, channelUrl }: DynamicParams,
+  { messagesDispatcher, logger }: StaticParams,
+): void {
+  const [inProgress, setInProgress] = useState(false);
+  useEffect(() => {
+    if (inProgress) {
+      return;
+    }
+    setInProgress(true);
+    setTimeout(() => {
+      if (typeof messagesLength === 'number'
+        && messagesLength > experimentalMessageLimit) {
+          logger.info('Trimming MessageList');
+          messagesDispatcher({
+            type: messageActionTypes.TRIM_MESSAGE_LIST,
+            payload: { experimentalMessageLimit },
+          });
```

Comment:
I think it's good to take messagesDispatcher from throttling logic
It will be called after 3secs, and that's different from what I thought